### ISSUE TITLE
Migrate to gRPC

### ISSUE THREAD

**Issue by m-novikov** [ilastik/tiktorch#71]
We should use [gRPC](https://github.com/grpc/grpc/) for our RPC.

### Why?
Pros:
- Supported in all programming languages, if somebody wants to connect to tiktorch server from java it would be easy to implement.
- It uses http2 for communication, this means we only need to have port 80/443 open on the server machine.
- Provides auth, encryption, and compression out of the box
- Allows data streaming (useful for such use-cases as live memory usage, loss function, number of iteration passed)

Cons:
- Lower performance than fine-tuned ZMQ implementation
- Requires grpc/protobuf dependency (protobuf already present in ilastik)

### What's wrong with our current setup (zmq)?
While it provided us good insight on how we want our RPC to look like, it's not the goal of this project to implement yet another RPC protocol.

### Why not plain http?
It allows define strict types (protobuf) for communication, no manual parsing required.
Easy client generation for multiple languages.

**Comment by m-novikov** [comment]
Resolved
